dif-data-classes/src/commonMain/kotlin/at/asitplus/dif/rqes/CollectionEntries/Document.kt

```diff
@@ -0,0 +1,87 @@
+package at.asitplus.dif.rqes.CollectionEntries
+
+import at.asitplus.dif.rqes.Enums.ConformanceLevelEnum
+import at.asitplus.dif.rqes.Enums.SignatureFormat
+import at.asitplus.dif.rqes.Enums.SignedEnvelopeProperty
+import at.asitplus.signum.indispensable.asn1.ObjectIdentifier
+import kotlinx.serialization.EncodeDefault
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * CSC: Class used as part of [SignatureRequestParameters]
+ */
+@Serializable
+data class Document(
+    /**
+     * base64-encoded document content to be signed, testcases weird so for now string
+     */
+    @SerialName("document")
+//    @Serializable(ByteArrayBase64Serializer::class)
+    val document: String,
+
+    /**
+     * Requested Signature Format
+     */
+    @SerialName("signature_format")
+    val signatureFormat: SignatureFormat,
+
+    /**
+     * Requested conformance level. If omitted its value is "Ades-B-B"
+     */
+    @SerialName("conformance_level")
+    val conformanceLevel: ConformanceLevelEnum? = null,
+
+    /**
+     * The OID of the algorithm to use for signing
+     */
+    @SerialName("signAlgo")
+    val signAlgo: ObjectIdentifier,
```

Comment:
Maybe replace name with `signatureAlgorithm`?

Reply:
`signatureAlgorithm` does not map to OIDs currently as noted at another point

Reply:
you can convert them to X509SignatureAlgorithms, they have an OID

Reply:
I have introduced an internal `signAlgorithm` which attempts to map the OID to our `SignatureAlgorithm` data class. See new commits